---
title: "Final Project"
author: "Kieckhaefer"
date: "October 27, 2015"
output: html_document
---

```{r eval = TRUE}
# Create table to link UCSC to RefSeq annotations

mm9.all = read.table("/Users/juliakieckhaefer/Desktop/Data Science/final-project/mm9geneID.txt", quote=NULL, header = F, sep = "\t")
head(mm9.all)
dim(mm9.all)
mm9 <- data.frame(mm9.all$V1, mm9.all$V2, mm9.all$V5, mm9.all$V6)
colnames(mm9) <- c("Transcript", "GeneID", "GeneName", "RefSeq")
head(mm9)
dim(mm9)
write.table(mm9, "/Users/juliakieckhaefer/Desktop/Data Science/final-project/mm9complete.txt", sep = "\t")

```

```{r eval = TRUE}
mouse.HUM = read.table("/Users/juliakieckhaefer/Desktop/Data Science/final-project/HOM_MouseHumanSequence.txt", quote=NULL, header = T, sep = "\t")
head(mouse.HUM)
tail(mouse.HUM)
even_indexes<-seq(2,40238,2)
odd_indexes<-seq(1,40239,2)
human <- data.frame(mouse.HUM[even_indexes, 1], mouse.HUM[even_indexes, 4])
colnames(human) <-c("index", "Human_name")
mouse <- data.frame(mouse.HUM[odd_indexes, 1], mouse.HUM[odd_indexes, 4])
colnames(mouse) <-c("index", "Mouse_name")
mouseTOhuman <- merge(human, mouse, by = "index")
```


```{r eval = TRUE}
RNAseq = read.table("/Users/juliakieckhaefer/Desktop/Data Science/final-project/Compare.txt", header = T, sep = "\t")
head(RNAseq)

library(dplyr)

RNAseq.gct <-data.frame(RNAseq$Gene, RNAseq$Transcript, RNAseq$Best, RNAseq$X462P, RNAseq$X1850P, RNAseq$X18294P, RNAseq$X1847P, RNAseq$X1088P, RNAseq$X20699P)
colnames(RNAseq.gct) <- c("Mouse_name", "Transcript", "Best", "X462P", "X1850P", "X18294P", "X1847P", "X1088P", "X20699P")
head(RNAseq.gct)

RNAseq.gct <- merge(mouseTOhuman, RNAseq.gct, by = "Mouse_name")

RNAseq.gct <- filter(RNAseq.gct, Best == 1)
head(RNAseq.gct)  
RNAseq.gct$Best <- NULL
RNAseq.gct$Mouse_name <- NULL
RNAseq.gct$index <- NULL
head(RNAseq.gct) 
write.table(RNAseq.gct, "/Users/juliakieckhaefer/Desktop/Data Science/final-project/RNAseqGCT.txt", quote = FALSE, sep = "\t", row.names = FALSE)
dim(RNAseq.gct)

```

```{r eval = TRUE}
# Clean and select relevant data from total RNAseq data produced by NGSC. Export to text file.

RNAseq = read.table("/Users/juliakieckhaefer/Desktop/Data Science/final-project/RNAseqUCSC.txt", header = T, sep = "\t")
head(RNAseq)
RNAseq.small <- data.frame(RNAseq$Gene, RNAseq$Transcript, RNAseq$Best, RNAseq$EDGE.M.log2.Mouse_Foxa1a2.def_Proximal.Mouse_Control_Proximal, RNAseq$EDGE.FDR.0.1.Mouse_Foxa1a2.def_Proximal.Mouse_Control_Proximal)
head(RNAseq.small)
colnames(RNAseq.small) <- c("Gene","Transcript", "Best.transcript", "Log2.ratio", "FDR")

RNAseq.ref <-merge(mm9, RNAseq.small, by = "Transcript")
head(RNAseq.ref)

library(dplyr)

RNAseq.sig <- RNAseq.ref %>%
  filter(FDR <= 0.02)
head(RNAseq.sig)

write.table(RNAseq.sig, "/Users/juliakieckhaefer/Desktop/Data Science/final-project/RNAsig02FDR.txt", row.names = FALSE, sep = "\t")


```




```{r eval = TRUE}
# Clean and import total ChIPseq data produced by NGSC. Export to text file. Generate .bed file and export.

ChIPseq = read.table("/Users/juliakieckhaefer/Desktop/Data Science/final-project/ChIPseq.txt", header = T, sep = "\t")
head(ChIPseq)
#Remove empty columns
ChIPseq$Span.Strand <- NULL
ChIPseq$X<- NULL
ChIPseq$X.1 <- NULL
ChIP.complete <- ChIPseq[complete.cases(ChIPseq), ]
write.table(ChIP.complete, "/Users/juliakieckhaefer/Desktop/Data Science/final-project/ChIPcomplete.txt", row.names = FALSE, sep = "\t")
head(ChIP.complete)


```

```{r eval=TRUE}

# Intersect of clean ChIPseq and RNAseq

combined <- merge(RNAseq.sig, ChIP.complete, by.x = "RefSeq", by.y = "Accession.number")
# Add column with absolute distance of the center of annotated peak from TSS.
combined <- cbind(combined, absTSSdist =  abs(combined$Gene.BeginBp-((combined$Span.EndBp-combined$Span.BeginBp)/2)))
head(combined)

library(ggplot2)
plot(combined$absTSSdist, combined$Log2.ratio)
plot(combined$absTSSdist, combined$FDR)

# Export combined data text file
write.table(combined, "/Users/juliakieckhaefer/Desktop/Data Science/final-project/boundRNAseq02FDR.txt", sep = "\t")
```

```{r eval = TRUE}
# Import RPM values for cleaned ChIP-seq data

RPM = read.table("/Users/juliakieckhaefer/Desktop/Data Science/final-project/AllPeaksCountsMinusD2408.txt", header = T, sep = "\t")
head(RPM)

RPM$strand <- NULL

plot(RPM$Foxa1Avg, RPM$Foxa2Avg)

ChIP.RPM <-merge(RPM, ChIP.complete, by = "Span.Name")
head(ChIP.RPM)

master <- merge(ChIP.RPM, RNAseq.sig, by = "Accession.number")
head(master)
plot(master$Foxa1Avg, master$Log2.ratio)
plot(master$Foxa2Avg, master$Log2.ratio)
plot(master$Foxa1Avg, master$AbsDistanceBp)
plot(master$Foxa1Avg, master$width)
plot(master$Foxa1Avg, master$Foxa2Avg)
plot(master$Foxa1Avg, master$FDR)



```
